src/main/cpp/sim/run_stride.cpp

```diff
@@ -42,34 +47,54 @@ using namespace std::chrono;
 using namespace checkpoint;
 
 std::mutex StrideSimulatorResult::io_mutex;
+bool load = false;
+boost::gregorian::date date;
 
-// temporary "global" for demo and testing purposes
-// CheckPoint* cp;
+#if USE_HDF5
+std::unique_ptr<CheckPoint> cp;
+#endif
 
 /// Performs an action just before a simulator step is performed.
-void StrideSimulatorResult::BeforeSimulatorStep(const Simulator& sim)
+void StrideSimulatorResult::BeforeSimulatorStep(Simulator& sim)
 {
 	run_clock.Start();
-	// saves the start configuration
-	/*
-	if (day == 0) {
-		cp->OpenFile();
-		cp->SaveCheckPoint(sim);
-		cp->CloseFile();
+
+#if USE_HDF5
+	if (stride::util::HDF5) {
+		if (load and day == 0) {
```

Comment:
These keyword aliases for `&&`, `||` and `!` are cute, but not all C++ compilers support them (Visual C++ is a notable example), and they don't appear anywhere else in our code. Can you perform the following replacements?

```cpp
and -> &&
or -> ||
not -> !
```